build-release-tools/application/reprove.py

```diff
@@ -396,6 +399,42 @@ def _update_dependency(self, version, pkg_version=None):
 
         return version
 
+
+    def create_tag(self):
+        if self._tag_name is None:
+            raise ValueError("No setting for branch-name")
+        else:
+            print "create tag for the repos..."
+            self.tag_existing_repositories()
+
+    def tag_existing_repositories(self):
+        """
+        Issues set_tagname commands to repos in a provided manifest
+        :return: None
+        """
+        repo_list = self._manifest.repositories
+        if repo_list is None:
+            print "No repository list found in manifest file"
+            sys.exit(2)
+        else:
+            # Loop through list of repos and create specified tag on each
+            for repo in repo_list:
+                self.set_repo_tagname(repo)
+  
+    def set_repo_tagname(self, repo):
+        """
+        Sets tagname on the repos in the manifest file
```

Comment:
this function seems to set single repo's tag, not all the repos in the manifest file